integration_tests/src/main/python/conftest.py

```diff
@@ -154,7 +154,9 @@ def pytest_runtest_setup(item):
     _inject_oom = item.get_closest_marker('inject_oom')
     datagen_overrides = item.get_closest_marker('datagen_overrides')
     if datagen_overrides:
-        _test_datagen_random_seed = datagen_overrides.kwargs.get('seed', _test_datagen_random_seed)
+        override_seed = datagen_overrides.kwargs.get('condition', True)
+        if override_seed:
+            _test_datagen_random_seed = datagen_overrides.kwargs.get('seed', _test_datagen_random_seed)
```

Comment:
Sort of a bug in the original code, but it's replicated here.  If the `datagen_overrides` marker is present, do we really want to ignore the fact that a seed kwarg was not found?  That probably indicates a typo, because the whole point of the datagen_overrides marker is to override the seed.

Reply:
Yeah, it doesn't make sense to have a `datagen_overrides` without a seed. I just carried over what was already done and added a condition to it. 

We can make the seed an arg instead of a kwarg.

Reply:
PTAL @jlowe I kept it kwarg as it's easier on the eyes to have a named parameter but made it mandatory